cirq/google/api/v1/params.py

```diff
@@ -49,58 +50,58 @@ def _to_zip_product(sweep: sweeps.Sweep) -> sweeps.Product:
     return sweep
 
 
-def _sweep_zip_to_proto_dict(sweep: sweeps.Zip) -> Dict:
+def _sweep_zip_to_proto(sweep: sweeps.Zip) -> params_pb2.ZipSweep:
     sweep_list = [
-        _single_param_sweep_to_proto_dict(cast(sweeps.SingleSweep, s))
+        _single_param_sweep_to_proto(cast(sweeps.SingleSweep, s))
         for s in sweep.sweeps
     ]
-    return {'sweeps': sweep_list}
+    return params_pb2.ZipSweep(sweeps=sweep_list)
 
 
-def _single_param_sweep_to_proto_dict(sweep: sweeps.SingleSweep) -> Dict:
-    msg = {}  # type: Dict
-    msg['parameter_key'] = sweep.key
+def _single_param_sweep_to_proto(sweep: sweeps.SingleSweep
+                                ) -> params_pb2.SingleSweep:
     if isinstance(sweep, sweeps.Linspace):
-        msg['linspace'] = {
-            'first_point': sweep.start,
-            'last_point': sweep.stop,
-            'num_points': sweep.length
-        }
+        return params_pb2.SingleSweep(parameter_key=sweep.key,
+                                      linspace=params_pb2.Linspace(
+                                          first_point=sweep.start,
+                                          last_point=sweep.stop,
+                                          num_points=sweep.length))
     elif isinstance(sweep, sweeps.Points):
-        msg['points'] = {'points': sweep.points}
+        return params_pb2.SingleSweep(
+            parameter_key=sweep.key,
+            points=params_pb2.Points(points=sweep.points))
     else:
         raise ValueError('invalid single-parameter sweep: {}'.format(sweep))
-    return msg
 
 
-def sweep_from_proto_dict(param_sweep: Dict) -> sweeps.Sweep:
-    if 'sweep' in param_sweep and 'factors' in param_sweep['sweep']:
+def sweep_from_proto(param_sweep: params_pb2.ParameterSweep) -> sweeps.Sweep:
+    if param_sweep.HasField('sweep') and len(param_sweep.sweep.factors) > 0:
```

Comment:
The serialization functionality should be equivalent. Swapping from REST to gRPC was a big enough change that I've tried to avoid making any other changes to behavior. If you're asking if the method should optimize the factors = 1 case to return _sweep_from_param_sweep_zip_proto(param_sweep.sweep.factors[0]) then possibly? Perhaps a future PR.

Reply:
I mean, if factors=1, should it even be a zip?  Shouldn't it then just be a singlesweep of points or linspace?

Reply:
I agree, but in an attempt to limit this PR I'm aiming for being functionally the same. A new small PR should probably exist to remove the unneeded sweeps.